File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        seen = {}

        for i in range(len(nums)):
            seen[nums[i]] = i
        
        for i in range(len(nums)):
            difference = target - nums[i]

            if difference in seen:
                value = seen[difference]
                if value and value != i:
                    return [min(i, value), max(i,value)]
        
        

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {

    unordered_map <int,int> dict;

    for (int i=0; i<nums.size(); i++){
        if (dict[nums[i]] == 1){
            return true;
        }
        else {
            dict[nums[i]] = 1;
        }
    }
    return false;
        
    }
};